src/runtime_src/core/common/device.cpp

```diff
@@ -102,6 +119,89 @@ get_axlf_section_or_error(axlf_section_kind section, const uuid& xclbin_id) cons
   throw std::runtime_error("no such xclbin section");
 }
 
+std::pair<const char*, size_t>
+device::
+get_axlf_section(const std::string& filename, axlf_section_kind kind) const
+{
```

Comment:
Who will use this API?  And why is an member function of xrt_core::device, it doesn't seem to be related to device at all?

Reply:
Yes, you're right. This and the next function aren't related to the device at all. My bad. Both of these functions are used by "SubCmdValidate.cpp" and "xbmgmt2/flash/firmware_image.cpp". Right now I created a copy and put it in here (firmware_image has its own copy of these functions), but I need to put them in a common place where those files can access these functions. 
I believe they belong to XBUtilities?